code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,278 @@
+		var/datum/surgery_step/current_step = current_surgery.get_surgery_step()
+		if(current_step.try_op(user, target, user.zone_selected, parent, current_surgery))
+			return
+		if(istype(parent, /obj/item/scalpel/laser/manager/debug))
+			return
+		if((can_cancel_before_first && current_surgery.status == 1) || current_surgery.status > 1)
+			attempt_cancel_surgery(current_surgery, target, user)
+			return
+
+	if(!isnull(current_surgery) && current_surgery.step_in_progress)
+		return
+
+	var/list/available_surgeries = get_available_surgeries(user, target)
+
+
+	var/datum/surgery/procedure
+
+	if(!length(available_surgeries))
+		if(IS_HORIZONTAL(target))
+			to_chat(user, "<span class='notice'>There aren't any surgeries you can perform there right now.</span>")
+		else
+			to_chat(user, "<span class='notice'>You can't perform any surgeries there while [target] is standing.</span>")
+		return
+
+	// if we have a surgery that should be performed regardless with this item,
+	// make sure it's available to be done
+	if(forced_surgery)
+		for(var/datum/surgery/S in available_surgeries)
+			if(istype(S, forced_surgery))
+				procedure = S
+				break
+	else
+		procedure = input("Begin which procedure?", "Surgery", null, null) as null|anything in available_surgeries
+
+	if(!procedure)
+		return
+
+	return try_choose_surgery(user, target, procedure)
+
+
+/datum/component/surgery_initiator/proc/get_available_surgeries(mob/user, mob/living/target)
+	var/list/available_surgeries = list()
+
+	var/mob/living/carbon/carbon_target
+	var/obj/item/organ/external/affecting
+	if (iscarbon(target))
+		carbon_target = target
+		affecting = carbon_target.get_organ(check_zone(user.zone_selected))
+
+	for(var/datum/surgery/surgery as anything in GLOB.surgeries_list)
+		if(surgery.abstract)  // no choosing abstract surgeries
+			continue
+		if(!surgery.possible_locs.Find(user.zone_selected))
+			continue
+		if(affecting)
+			if(!surgery.requires_bodypart)
+				continue
+			if((surgery.requires_organic_bodypart && affecting.is_robotic()) || (!surgery.requires_organic_bodypart && !affecting.is_robotic()))
+				continue
+			if(surgery.requires_real_bodypart && !affecting.is_primary_organ())
+				continue
+		else if(carbon_target && surgery.requires_bodypart) //mob with no limb in surgery zone when we need a limb
+			continue
+		if(surgery.lying_required && target.body_position != LYING_DOWN)
+			continue
+		if(!surgery.self_operable && target == user)
+			continue
+		if(!surgery.can_start(user, target))
+			continue
+		for(var/path in surgery.target_mobtypes)
+			if(istype(target, path))
+				available_surgeries += surgery
+				break
+
+	return available_surgeries
+
+/// Does the surgery de-initiation.
+/datum/component/surgery_initiator/proc/attempt_cancel_surgery(datum/surgery/the_surgery, mob/living/patient, mob/user)
+	var/selected_zone = user.zone_selected
+	/// We haven't even started yet. Any surgery can be cancelled at this point.
+	if(the_surgery.status == 1)
```

Comment:
same here, is this supposed to be true or some other define?

Reply:
Addressed in previous comment